USFMToolsSharp.Renderers.Docx/DocxRenderer.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using USFMToolsSharp.Models.Markers;
+using NPOI.XWPF.UserModel;
+using System.IO;
+
+namespace USFMToolsSharp.Renderers.Docx
+{
+    public class DocxRenderer
+    {
+        public List<string> UnrenderableTags;
+        public Dictionary<string,List<Marker>> FootnoteTextTags;
+        
+        private XWPFDocument newDoc;
+
+        private bool isSingleSpaced = true;
+        private bool hasOneColumn = true;
+        private bool isL2RDirection = true;
+        private bool isTextJustified = false;
+        private bool separateChapters = true;
+
+        public string FrontMatterDocx { get; set; }
+        public string InsertedFooter { get; set; }
+        public string InsertedHead { get; set; }
+
+        public DocxRenderer()
+        {
+            UnrenderableTags = new List<string>();
+            FootnoteTextTags = new Dictionary<string,List<Marker>>();
+            newDoc = new XWPFDocument();
+        }
+        public DocxRenderer(bool isSingleSpaced = true, bool hasOneColumn = true, bool isL2RDirection = true, bool isTextJustified = false, bool separateChapters = false)
+        {
+            this.isSingleSpaced = isSingleSpaced;
+            this.hasOneColumn = hasOneColumn;
+            this.isL2RDirection = isL2RDirection;
+            this.isTextJustified = isTextJustified;
+            this.separateChapters = separateChapters;
+
+            UnrenderableTags = new List<string>();
+            FootnoteTextTags = new Dictionary<string, List<Marker>>();
+            newDoc = new XWPFDocument();
+
+        }
+
+
+        public XWPFDocument Render(USFMDocument input)
+        {
+
+                foreach (Marker marker in input.Contents)
+                {
+
+                    RenderMarker(marker);
+
+                }
+            return newDoc;
+
+        }
+        private void RenderMarker(Marker input, XWPFParagraph parentParagraph = null, bool isBold = false, bool isItalics = false)
+        {
+            
+            switch (input)
+            {
+                case PMarker _:
+                    
+                        XWPFParagraph newParagraph = newDoc.CreateParagraph();
+                        foreach (Marker marker in input.Contents)
+                        {
+                            RenderMarker(marker, newParagraph);
+                        }
+                    break;
+                case CMarker cMarker:
+                    XWPFParagraph newChapter = newDoc.CreateParagraph();
+                    XWPFRun chapterMarker = newChapter.CreateRun();
+                    chapterMarker.SetText($"{cMarker.Number}");
```

Comment:
You could have just used cMarker.Number.ToString() but this will optimize down to that